source/common/http/http1/codec_impl.cc

```diff
@@ -187,7 +188,31 @@ void StreamEncoderImpl::encodeData(Buffer::Instance& data, bool end_stream) {
   }
 }
 
-void StreamEncoderImpl::encodeTrailers(const HeaderMap&) { endEncode(); }
+void StreamEncoderImpl::encodeTrailers(const HeaderMap& trailers) {
+
+  // Trailers only matter if it is a chunk transfer encoding
+  // https://tools.ietf.org/html/rfc7230#section-4.4
+  if (chunk_encoding_) {
+    // Finalize the body
+    connection_.buffer().add(LAST_CHUNK);
+
+    trailers.iterate(
+        [](const HeaderEntry& header, void* context) -> HeaderMap::Iterate {
+          static_cast<StreamEncoderImpl*>(context)->encodeFormattedHeader(
+              header.key().getStringView(), header.value().getStringView());
+          return HeaderMap::Iterate::Continue;
+        },
+        this);
+
+    // We have to flush output here since we use encodeFormattedHeader?
+    // Not sure why
+    connection_.flushOutput();
```

Comment:
@mattklein123 Got into a weird situation where I had to flush the output here otherwise I get an assertion error later.

Reply:
Because of the way encode header works. It reserves data into an intermediate buffer for perf reasons. Take a look at the code. Please update the comment.